Python/optimizer.c

```diff
@@ -800,24 +774,6 @@ translate_bytecode_to_trace(
                 2 * INSTR_IP(initial_instr, code),
                 trace_length,
                 buffer_size - max_length);
-        if (max_length < buffer_size) {
-            // There are stubs
-            if (trace_length < max_length) {
-                // There's a gap before the stubs
-                // Move the stubs back to be immediately after the main trace
-                // (which ends at trace_length)
-                DPRINTF(2,
-                        "Moving %d stub uops back by %d\n",
-                        buffer_size - max_length,
-                        max_length - trace_length);
-                trace_length = move_stubs(trace, trace_length, max_length, buffer_size);
-            }
-            else {
-                assert(trace_length == max_length);
-                // There's no gap
-                trace_length = buffer_size;
-            }
-        }
         return trace_length;
```

Comment:
Note that the meaning of the return value is changed: previously it would be the number of instructions including stubs; now it's number of instructions *without* counting stubs. Maybe add a comment to the top of the function explaining the meaning. (It also took me several attempts to understand that it *must* return this value, since that's needed for `remove_unneeded_uops()`.)

Also, the buffer_length parameter is not strictly needed any more -- it could just be a local variable initialized to `_Py_UOP_MAX_TRACE_LENGTH`, since the assumption that the buffer has that size is now baked in elsewhere.

Reply:
I've changed this to just return 1 for success, instead of the length.